### ISSUE TITLE
vMX 20.1R1.11 shows unhealthy state but vrcons working fine.

### ISSUE THREAD

**Issue by kalmanoharan** [vrnetlab/vrnetlab#229]
Hi @plajjan ,
I am trying to use vMX 20.1R1.11 with vrnet lab and noticed that `docker ps` shows the container unhealthy. But if I try to look at both docker logs and vrcons into the router. Everything looks normal. Just wondering why docker still assumes that container is unhealthy and how to fix this ?

Here are the logs for your reference.
```
root@nel17-lab1-vagrantserver1:/home/kalmanoharan/vrnetlab# docker ps
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                   PORTS                                                 NAMES
f62043535551        vrnetlab/vr-vmx:20.1R1.11   "/launch.py"             2 hours ago         Up 2 hours (unhealthy)   22/tcp, 830/tcp, 5000/tcp, 10000-10099/tcp, 161/udp   R2
3bdcb8af6996        vrnetlab/vr-vmx:20.1R1.11   "/launch.py"             2 hours ago         Up 2 hours (unhealthy)   22/tcp, 830/tcp, 5000/tcp, 10000-10099/tcp, 161/udp   R1
```
```
root@nel17-lab1-vagrantserver1:/home/kalmanoharan/vrnetlab# docker logs 3bdcb8af6996
2020-06-01 07:52:23,288: vrnetlab   DEBUG    Creating overlay disk image
2020-06-01 07:52:23,301: vrnetlab   DEBUG    Creating overlay disk image
2020-06-01 07:52:23,315: vrnetlab   DEBUG    Starting vrnetlab VMX
2020-06-01 07:52:23,315: vrnetlab   DEBUG    VMs: [<__main__.VMX_vcp object at 0x7faf96fb05c0>, <__main__.VMX_vfpc object at 0x7faf96fb0710>]
2020-06-01 07:52:23,326: vrnetlab   DEBUG    VM not started; starting!
2020-06-01 07:52:23,327: vrnetlab   INFO     Starting VMX_vcp
2020-06-01 07:52:23,327: vrnetlab   DEBUG    ['qemu-system-x86_64', '-enable-kvm', '-display', 'none', '-machine', 'pc', '-monitor', 'tcp:0.0.0.0:4000,server,nowait', '-m', '2048', '-serial', 'telnet:0.0.0.0:5000,server,nowait', '-drive', 'if=ide,file=/vmx/junos-vmx-x86-64-20.1R1.11-overlay-overlay.qcow2', '-drive', 'if=ide,file=/vmx/vmxhdd.img', '-usb', '-usbdevice', 'disk:format=raw:/vmx/metadata-usb-re.img', '-smbios', 'type=0,vendor=Juniper', '-smbios', 'type=1,manufacturer=Juniper,product=VM-vcp_vmx2-161-re-0,version=0.1.0', '-device', 'e1000,netdev=p00,mac=52:54:00:8a:7a:00', '-netdev', 'user,id=p00,net=10.0.0.0/24,tftp=/tftpboot,hostfwd=tcp::2022-10.0.0.15:22,hostfwd=udp::2161-10.0.0.15:161,hostfwd=tcp::2830-10.0.0.15:830,hostfwd=tcp::2080-10.0.0.15:80,hostfwd=tcp::2443-10.0.0.15:443', '-device', 'virtio-net-pci,netdev=vcp-int,mac=52:54:00:8b:31:01', '-netdev', 'tap,ifname=vcp-int,id=vcp-int,script=no,downscript=no']
2020-06-01 07:52:29,342: vrnetlab   DEBUG    VM not started; starting!
2020-06-01 07:52:29,342: vrnetlab   INFO     Starting VMX_vfpc
2020-06-01 07:52:29,347: vrnetlab   DEBUG    ['qemu-system-x86_64', '-enable-kvm', '-display', 'none', '-machine', 'pc', '-monitor', 'tcp:0.0.0.0:4001,server,nowait', '-m', '4096', '-serial', 'telnet:0.0.0.0:5001,server,nowait', '-drive', 'if=ide,file=/vmx/vfpc-overlay.img', '-cpu', 'SandyBridge', '-M', 'pc', '-smp', '3', '-usb', '-usbdevice', 'disk:format=raw:/vmx/metadata-usb-fpc0.img', '-device', 'pci-bridge,chassis_nr=1,id=pci.1', '-device', 'pci-bridge,chassis_nr=2,id=pci.2', '-device', 'pci-bridge,chassis_nr=3,id=pci.3', '-device', 'pci-bridge,chassis_nr=4,id=pci.4', '-device', 'virtio-net-pci,netdev=mgmt,mac=52:54:00:83:79:00', '-netdev', 'user,id=mgmt,net=10.0.0.0/24', '-device', 'virtio-net-pci,netdev=vfpc-int,mac=52:54:00:72:02:00', '-netdev', 'tap,ifname=vfpc-int,id=vfpc-int,script=no,downscript=no', '-device', 'virtio-net-pci,netdev=p01,mac=52:54:00:59:b1:01,bus=pci.1,addr=0x2', '-netdev', 'socket,id=p01,listen=:10001', '-device', 'virtio-net-pci,netdev=p02,mac=52:54:00:9f:8d:02,bus=pci.1,addr=0x3', '-netdev', 'socket,id=p02,listen=:10002', '-device', 'virtio-net-pci,netdev=p03,mac=52:54:00:17:2b:03,bus=pci.1,addr=0x4', '-netdev', 'socket,id=p03,listen=:10003', '-device', 'virtio-net-pci,netdev=p04,mac=52:54:00:a8:d8:04,bus=pci.1,addr=0x5', '-netdev', 'socket,id=p04,listen=:10004', '-device', 'virtio-net-pci,netdev=p05,mac=52:54:00:bc:e1:05,bus=pci.1,addr=0x6', '-netdev', 'socket,id=p05,listen=:10005', '-device', 'virtio-net-pci,netdev=p06,mac=52:54:00:31:9b:06,bus=pci.1,addr=0x7', '-netdev', 'socket,id=p06,listen=:10006', '-device', 'virtio-net-pci,netdev=p07,mac=52:54:00:20:d5:07,bus=pci.1,addr=0x8', '-netdev', 'socket,id=p07,listen=:10007', '-device', 'virtio-net-pci,netdev=p08,mac=52:54:00:42:ca:08,bus=pci.1,addr=0x9', '-netdev', 'socket,id=p08,listen=:10008', '-device', 'virtio-net-pci,netdev=p09,mac=52:54:00:7e:e5:09,bus=pci.1,addr=0xa', '-netdev', 'socket,id=p09,listen=:10009', '-device', 'virtio-net-pci,netdev=p10,mac=52:54:00:15:b8:0a,bus=pci.1,addr=0xb', '-netdev', 'socket,id=p10,listen=:10010', '-device', 'virtio-net-pci,netdev=p11,mac=52:54:00:26:84:0b,bus=pci.1,addr=0xc', '-netdev', 'socket,id=p11,listen=:10011', '-device', 'virtio-net-pci,netdev=p12,mac=52:54:00:3f:2d:0c,bus=pci.1,addr=0xd', '-netdev', 'socket,id=p12,listen=:10012', '-device', 'virtio-net-pci,netdev=p13,mac=52:54:00:0e:b2:0d,bus=pci.1,addr=0xe', '-netdev', 'socket,id=p13,listen=:10013', '-device', 'virtio-net-pci,netdev=p14,mac=52:54:00:c9:74:0e,bus=pci.1,addr=0xf', '-netdev', 'socket,id=p14,listen=:10014', '-device', 'virtio-net-pci,netdev=p15,mac=52:54:00:b0:be:0f,bus=pci.1,addr=0x10', '-netdev', 'socket,id=p15,listen=:10015', '-device', 'virtio-net-pci,netdev=p16,mac=52:54:00:36:03:10,bus=pci.1,addr=0x11', '-netdev', 'socket,id=p16,listen=:10016', '-device', 'virtio-net-pci,netdev=p17,mac=52:54:00:e6:73:11,bus=pci.1,addr=0x12', '-netdev', 'socket,id=p17,listen=:10017', '-device', 'virtio-net-pci,netdev=p18,mac=52:54:00:15:b6:12,bus=pci.1,addr=0x13', '-netdev', 'socket,id=p18,listen=:10018', '-device', 'virtio-net-pci,netdev=p19,mac=52:54:00:82:e4:13,bus=pci.1,addr=0x14', '-netdev', 'socket,id=p19,listen=:10019', '-device', 'virtio-net-pci,netdev=p20,mac=52:54:00:be:0a:14,bus=pci.1,addr=0x15', '-netdev', 'socket,id=p20,listen=:10020', '-device', 'virtio-net-pci,netdev=p21,mac=52:54:00:6a:a4:15,bus=pci.1,addr=0x16', '-netdev', 'socket,id=p21,listen=:10021', '-device', 'virtio-net-pci,netdev=p22,mac=52:54:00:ea:d0:16,bus=pci.1,addr=0x17', '-netdev', 'socket,id=p22,listen=:10022', '-device', 'virtio-net-pci,netdev=p23,mac=52:54:00:af:38:17,bus=pci.1,addr=0x18', '-netdev', 'socket,id=p23,listen=:10023', '-device', 'virtio-net-pci,netdev=p24,mac=52:54:00:21:62:18,bus=pci.1,addr=0x19', '-netdev', 'socket,id=p24,listen=:10024', '-device', 'virtio-net-pci,netdev=p25,mac=52:54:00:5b:d2:19,bus=pci.1,addr=0x1a', '-netdev', 'socket,id=p25,listen=:10025', '-device', 'virtio-net-pci,netdev=p26,mac=52:54:00:d5:61:1a,bus=pci.2,addr=0x1', '-netdev', 'socket,id=p26,listen=:10026', '-device', 'virtio-net-pci,netdev=p27,mac=52:54:00:1d:69:1b,bus=pci.2,addr=0x2', '-netdev', 'socket,id=p27,listen=:10027', '-device', 'virtio-net-pci,netdev=p28,mac=52:54:00:c0:9c:1c,bus=pci.2,addr=0x3', '-netdev', 'socket,id=p28,listen=:10028', '-device', 'virtio-net-pci,netdev=p29,mac=52:54:00:cf:59:1d,bus=pci.2,addr=0x4', '-netdev', 'socket,id=p29,listen=:10029', '-device', 'virtio-net-pci,netdev=p30,mac=52:54:00:93:5c:1e,bus=pci.2,addr=0x5', '-netdev', 'socket,id=p30,listen=:10030', '-device', 'virtio-net-pci,netdev=p31,mac=52:54:00:8b:f0:1f,bus=pci.2,addr=0x6', '-netdev', 'socket,id=p31,listen=:10031', '-device', 'virtio-net-pci,netdev=p32,mac=52:54:00:f4:db:20,bus=pci.2,addr=0x7', '-netdev', 'socket,id=p32,listen=:10032', '-device', 'virtio-net-pci,netdev=p33,mac=52:54:00:3c:c7:21,bus=pci.2,addr=0x8', '-netdev', 'socket,id=p33,listen=:10033', '-device', 'virtio-net-pci,netdev=p34,mac=52:54:00:5e:c4:22,bus=pci.2,addr=0x9', '-netdev', 'socket,id=p34,listen=:10034', '-device', 'virtio-net-pci,netdev=p35,mac=52:54:00:79:fe:23,bus=pci.2,addr=0xa', '-netdev', 'socket,id=p35,listen=:10035', '-device', 'virtio-net-pci,netdev=p36,mac=52:54:00:f5:97:24,bus=pci.2,addr=0xb', '-netdev', 'socket,id=p36,listen=:10036', '-device', 'virtio-net-pci,netdev=p37,mac=52:54:00:d8:f6:25,bus=pci.2,addr=0xc', '-netdev', 'socket,id=p37,listen=:10037', '-device', 'virtio-net-pci,netdev=p38,mac=52:54:00:99:07:26,bus=pci.2,addr=0xd', '-netdev', 'socket,id=p38,listen=:10038', '-device', 'virtio-net-pci,netdev=p39,mac=52:54:00:69:5a:27,bus=pci.2,addr=0xe', '-netdev', 'socket,id=p39,listen=:10039', '-device', 'virtio-net-pci,netdev=p40,mac=52:54:00:ef:30:28,bus=pci.2,addr=0xf', '-netdev', 'socket,id=p40,listen=:10040', '-device', 'virtio-net-pci,netdev=p41,mac=52:54:00:8c:2a:29,bus=pci.2,addr=0x10', '-netdev', 'socket,id=p41,listen=:10041', '-device', 'virtio-net-pci,netdev=p42,mac=52:54:00:6a:ba:2a,bus=pci.2,addr=0x11', '-netdev', 'socket,id=p42,listen=:10042', '-device', 'virtio-net-pci,netdev=p43,mac=52:54:00:0e:73:2b,bus=pci.2,addr=0x12', '-netdev', 'socket,id=p43,listen=:10043', '-device', 'virtio-net-pci,netdev=p44,mac=52:54:00:da:68:2c,bus=pci.2,addr=0x13', '-netdev', 'socket,id=p44,listen=:10044', '-device', 'virtio-net-pci,netdev=p45,mac=52:54:00:09:c8:2d,bus=pci.2,addr=0x14', '-netdev', 'socket,id=p45,listen=:10045', '-device', 'virtio-net-pci,netdev=p46,mac=52:54:00:0f:23:2e,bus=pci.2,addr=0x15', '-netdev', 'socket,id=p46,listen=:10046', '-device', 'virtio-net-pci,netdev=p47,mac=52:54:00:12:a0:2f,bus=pci.2,addr=0x16', '-netdev', 'socket,id=p47,listen=:10047', '-device', 'virtio-net-pci,netdev=p48,mac=52:54:00:32:f4:30,bus=pci.2,addr=0x17', '-netdev', 'socket,id=p48,listen=:10048', '-device', 'virtio-net-pci,netdev=p49,mac=52:54:00:ab:f6:31,bus=pci.2,addr=0x18', '-netdev', 'socket,id=p49,listen=:10049', '-device', 'virtio-net-pci,netdev=p50,mac=52:54:00:a9:80:32,bus=pci.2,addr=0x19', '-netdev', 'socket,id=p50,listen=:10050', '-device', 'virtio-net-pci,netdev=p51,mac=52:54:00:6e:02:33,bus=pci.2,addr=0x1a', '-netdev', 'socket,id=p51,listen=:10051', '-device', 'virtio-net-pci,netdev=p52,mac=52:54:00:4d:ca:34,bus=pci.3,addr=0x1', '-netdev', 'socket,id=p52,listen=:10052', '-device', 'virtio-net-pci,netdev=p53,mac=52:54:00:0f:e2:35,bus=pci.3,addr=0x2', '-netdev', 'socket,id=p53,listen=:10053', '-device', 'virtio-net-pci,netdev=p54,mac=52:54:00:4f:14:36,bus=pci.3,addr=0x3', '-netdev', 'socket,id=p54,listen=:10054', '-device', 'virtio-net-pci,netdev=p55,mac=52:54:00:89:ba:37,bus=pci.3,addr=0x4', '-netdev', 'socket,id=p55,listen=:10055', '-device', 'virtio-net-pci,netdev=p56,mac=52:54:00:95:49:38,bus=pci.3,addr=0x5', '-netdev', 'socket,id=p56,listen=:10056', '-device', 'virtio-net-pci,netdev=p57,mac=52:54:00:92:e1:39,bus=pci.3,addr=0x6', '-netdev', 'socket,id=p57,listen=:10057', '-device', 'virtio-net-pci,netdev=p58,mac=52:54:00:2e:e4:3a,bus=pci.3,addr=0x7', '-netdev', 'socket,id=p58,listen=:10058', '-device', 'virtio-net-pci,netdev=p59,mac=52:54:00:e9:d5:3b,bus=pci.3,addr=0x8', '-netdev', 'socket,id=p59,listen=:10059', '-device', 'virtio-net-pci,netdev=p60,mac=52:54:00:37:ab:3c,bus=pci.3,addr=0x9', '-netdev', 'socket,id=p60,listen=:10060', '-device', 'virtio-net-pci,netdev=p61,mac=52:54:00:08:61:3d,bus=pci.3,addr=0xa', '-netdev', 'socket,id=p61,listen=:10061', '-device', 'virtio-net-pci,netdev=p62,mac=52:54:00:11:e8:3e,bus=pci.3,addr=0xb', '-netdev', 'socket,id=p62,listen=:10062', '-device', 'virtio-net-pci,netdev=p63,mac=52:54:00:0e:29:3f,bus=pci.3,addr=0xc', '-netdev', 'socket,id=p63,listen=:10063', '-device', 'virtio-net-pci,netdev=p64,mac=52:54:00:97:6b:40,bus=pci.3,addr=0xd', '-netdev', 'socket,id=p64,listen=:10064', '-device', 'virtio-net-pci,netdev=p65,mac=52:54:00:b1:f2:41,bus=pci.3,addr=0xe', '-netdev', 'socket,id=p65,listen=:10065', '-device', 'virtio-net-pci,netdev=p66,mac=52:54:00:af:99:42,bus=pci.3,addr=0xf', '-netdev', 'socket,id=p66,listen=:10066', '-device', 'virtio-net-pci,netdev=p67,mac=52:54:00:1f:9b:43,bus=pci.3,addr=0x10', '-netdev', 'socket,id=p67,listen=:10067', '-device', 'virtio-net-pci,netdev=p68,mac=52:54:00:c1:2a:44,bus=pci.3,addr=0x11', '-netdev', 'socket,id=p68,listen=:10068', '-device', 'virtio-net-pci,netdev=p69,mac=52:54:00:47:5b:45,bus=pci.3,addr=0x12', '-netdev', 'socket,id=p69,listen=:10069', '-device', 'virtio-net-pci,netdev=p70,mac=52:54:00:ad:49:46,bus=pci.3,addr=0x13', '-netdev', 'socket,id=p70,listen=:10070', '-device', 'virtio-net-pci,netdev=p71,mac=52:54:00:61:39:47,bus=pci.3,addr=0x14', '-netdev', 'socket,id=p71,listen=:10071', '-device', 'virtio-net-pci,netdev=p72,mac=52:54:00:1e:31:48,bus=pci.3,addr=0x15', '-netdev', 'socket,id=p72,listen=:10072', '-device', 'virtio-net-pci,netdev=p73,mac=52:54:00:87:09:49,bus=pci.3,addr=0x16', '-netdev', 'socket,id=p73,listen=:10073', '-device', 'virtio-net-pci,netdev=p74,mac=52:54:00:38:b0:4a,bus=pci.3,addr=0x17', '-netdev', 'socket,id=p74,listen=:10074', '-device', 'virtio-net-pci,netdev=p75,mac=52:54:00:bc:f1:4b,bus=pci.3,addr=0x18', '-netdev', 'socket,id=p75,listen=:10075', '-device', 'virtio-net-pci,netdev=p76,mac=52:54:00:55:18:4c,bus=pci.3,addr=0x19', '-netdev', 'socket,id=p76,listen=:10076', '-device', 'virtio-net-pci,netdev=p77,mac=52:54:00:37:79:4d,bus=pci.3,addr=0x1a', '-netdev', 'socket,id=p77,listen=:10077', '-device', 'virtio-net-pci,netdev=p78,mac=52:54:00:75:1a:4e,bus=pci.4,addr=0x1', '-netdev', 'socket,id=p78,listen=:10078', '-device', 'virtio-net-pci,netdev=p79,mac=52:54:00:61:5a:4f,bus=pci.4,addr=0x2', '-netdev', 'socket,id=p79,listen=:10079', '-device', 'virtio-net-pci,netdev=p80,mac=52:54:00:d8:f2:50,bus=pci.4,addr=0x3', '-netdev', 'socket,id=p80,listen=:10080', '-device', 'virtio-net-pci,netdev=p81,mac=52:54:00:ad:2b:51,bus=pci.4,addr=0x4', '-netdev', 'socket,id=p81,listen=:10081', '-device', 'virtio-net-pci,netdev=p82,mac=52:54:00:f4:9c:52,bus=pci.4,addr=0x5', '-netdev', 'socket,id=p82,listen=:10082', '-device', 'virtio-net-pci,netdev=p83,mac=52:54:00:81:4b:53,bus=pci.4,addr=0x6', '-netdev', 'socket,id=p83,listen=:10083', '-device', 'virtio-net-pci,netdev=p84,mac=52:54:00:b8:6c:54,bus=pci.4,addr=0x7', '-netdev', 'socket,id=p84,listen=:10084', '-device', 'virtio-net-pci,netdev=p85,mac=52:54:00:d8:6b:55,bus=pci.4,addr=0x8', '-netdev', 'socket,id=p85,listen=:10085', '-device', 'virtio-net-pci,netdev=p86,mac=52:54:00:f7:78:56,bus=pci.4,addr=0x9', '-netdev', 'socket,id=p86,listen=:10086', '-device', 'virtio-net-pci,netdev=p87,mac=52:54:00:b2:0f:57,bus=pci.4,addr=0xa', '-netdev', 'socket,id=p87,listen=:10087', '-device', 'virtio-net-pci,netdev=p88,mac=52:54:00:13:50:58,bus=pci.4,addr=0xb', '-netdev', 'socket,id=p88,listen=:10088', '-device', 'virtio-net-pci,netdev=p89,mac=52:54:00:e7:5a:59,bus=pci.4,addr=0xc', '-netdev', 'socket,id=p89,listen=:10089', '-device', 'virtio-net-pci,netdev=p90,mac=52:54:00:10:a8:5a,bus=pci.4,addr=0xd', '-netdev', 'socket,id=p90,listen=:10090', '-device', 'virtio-net-pci,netdev=p91,mac=52:54:00:76:a3:5b,bus=pci.4,addr=0xe', '-netdev', 'socket,id=p91,listen=:10091', '-device', 'virtio-net-pci,netdev=p92,mac=52:54:00:2d:b9:5c,bus=pci.4,addr=0xf', '-netdev', 'socket,id=p92,listen=:10092', '-device', 'virtio-net-pci,netdev=p93,mac=52:54:00:a9:79:5d,bus=pci.4,addr=0x10', '-netdev', 'socket,id=p93,listen=:10093', '-device', 'virtio-net-pci,netdev=p94,mac=52:54:00:f7:e2:5e,bus=pci.4,addr=0x11', '-netdev', 'socket,id=p94,listen=:10094', '-device', 'virtio-net-pci,netdev=p95,mac=52:54:00:77:f9:5f,bus=pci.4,addr=0x12', '-netdev', 'socket,id=p95,listen=:10095', '-device', 'virtio-net-pci,netdev=p96,mac=52:54:00:9c:31:60,bus=pci.4,addr=0x13', '-netdev', 'socket,id=p96,listen=:10096']
2020-06-01 07:55:56,675: launch     INFO     matched login prompt
2020-06-01 07:55:56,676: launch     DEBUG    writing to serial console: root
2020-06-01 07:55:59,685: launch     INFO     matched login prompt
2020-06-01 07:55:59,686: launch     DEBUG    writing to serial console: root
2020-06-01 07:56:02,688: launch     DEBUG    writing to serial console: cli
2020-06-01 07:56:03,062: launch     DEBUG    writing to serial console: configure
2020-06-01 07:56:03,104: launch     DEBUG    writing to serial console: set chassis fpc 0 pic 0 number-of-ports 96
2020-06-01 07:56:03,145: launch     DEBUG    writing to serial console: set system services ssh
2020-06-01 07:56:03,186: launch     DEBUG    writing to serial console: set system services netconf ssh
2020-06-01 07:56:03,228: launch     DEBUG    writing to serial console: set system services netconf rfc-compliant
2020-06-01 07:56:03,269: launch     DEBUG    writing to serial console: set system login user vrnetlab class super-user authentication plain-text-password
2020-06-01 07:56:03,312: launch     DEBUG    writing to serial console: VR-netlab9
2020-06-01 07:56:03,354: launch     DEBUG    writing to serial console: VR-netlab9
2020-06-01 07:56:03,393: launch     DEBUG    writing to serial console: set system root-authentication plain-text-password
2020-06-01 07:56:03,432: launch     DEBUG    writing to serial console: VR-netlab9
2020-06-01 07:56:03,474: launch     DEBUG    writing to serial console: VR-netlab9
2020-06-01 07:56:03,512: launch     DEBUG    writing to serial console: set interfaces fxp0 unit 0 family inet address 10.0.0.15/24
2020-06-01 07:56:03,552: launch     DEBUG    writing to serial console: delete interfaces fxp0 unit 0 family inet dhcp
2020-06-01 07:56:03,594: launch     DEBUG    writing to serial console: delete system processes dhcp-service
2020-06-01 07:56:03,636: launch     DEBUG    writing to serial console: commit
2020-06-01 07:56:08,331: launch     DEBUG    writing to serial console: exit
2020-06-01 07:56:08,334: launch     INFO     Startup complete in: 0:03:45.007179
root@nel17-lab1-vagrantserver1:/home/kalmanoharan/vrnetlab#
```

```
root@nel17-lab1-vagrantserver1:/home/kalmanoharan/vrnetlab# vrcons R1
Trying 172.17.0.14...
Connected to 172.17.0.14.
Escape character is '^]'.


[edit]
root# run show interfaces terse
Interface               Admin Link Proto    Local                 Remote
ge-0/0/0                up    up
ge-0/0/0.16386          up    up
lc-0/0/0                up    up
lc-0/0/0.32769          up    up   vpls
pfe-0/0/0               up    up
pfe-0/0/0.16383         up    up   inet
                                   inet6
pfh-0/0/0               up    up
pfh-0/0/0.16383         up    up   inet
pfh-0/0/0.16384         up    up   inet
ge-0/0/1                up    up
ge-0/0/1.16386          up    up
ge-0/0/2                up    up
ge-0/0/2.16386          up    up
ge-0/0/3                up    up
ge-0/0/3.16386          up    up
ge-0/0/4                up    up
ge-0/0/4.16386          up    up
ge-0/0/5                up    up
ge-0/0/5.16386          up    up
ge-0/0/6                up    up
ge-0/0/6.16386          up    up
ge-0/0/7                up    up
ge-0/0/7.16386          up    up
ge-0/0/8                up    up
ge-0/0/8.16386          up    up
ge-0/0/9                up    up
ge-0/0/9.16386          up    up
cbp0                    up    up
demux0                  up    up
dsc                     up    up
em1                     up    up
em1.0                   up    up   inet     10.0.0.4/8
                                            128.0.0.1/2
                                            128.0.0.4/2
                                   inet6    fe80::5254:ff:fe8b:3101/64
                                            fec0::a:0:0:4/64
                                   tnp      0x4
esi                     up    up
fti0                    up    up
fti1                    up    up
fti2                    up    up
fti3                    up    up
fti4                    up    up
fti5                    up    up
```

Also wondering what are those .16386 subinterfaces are they are never defined on the configuratons.

**Comment by plajjan** [comment]
The healthcheck is a rather primitive one. I guess there is something wrong with it - does it ever become healthy? On some platforms I think it can take a really long time (20 minutes).

JUNOS doesn't require interfaces to be configured for them to be present on the device. I don't know what those interfaces are but I don't think it is related to vrnetlab - better to search juniper forums etc. If you do find out, you could open an issue (or submit code in a PR) here to document it for the next person wondering the same thing.

Would be interesting to see if other people had similar issues with the healthcheck on vmx.

**Comment by kalmanoharan** [comment]
Thanks @plajjan, For the feedback. I'll update here once I figured out the reason for it.

**Comment by mbound** [comment]
I can confirm I always had this same issue with vMX (never showed as "healthy")

**Comment by hellt** [comment]
Hi guys, I wonder if during the build of 20.1R1.11 you also had the auto-image upgrade as the last step?
Because my `make` hangs with the last messages being:

```
Broadcast Message from root@                                                   
        (no tty) at 14:39 UTC...                                               
                                                                               
ALERT:Auto image Upgrade will start. This can terminate config         CLI sess
ion(s). Modified configuration will be lost. To stop Auto-image, in CLI do the 
following: 'edit; delete chassis auto-image-upgrade; commit'.                  
                                                                               

error: session failure: unexpected termination
error: remote side unexpectedly closed connection
root@:~ #
2021-01-20 14:39:10,019: launch     DEBUG    writing to serial console: exit
2021-01-20 14:39:10,019: launch     INFO     Startup complete in: 0:16:50.806108
```

I wonder if this is a change with the `20.2.R1.10` version that I am trying to build?

**Comment by hellt** [comment]
I suppose this is due to the fact, that the image that is commited after the install phase has the `--trace --install` arguments in CMD

result of `image inspect`
```
     "Cmd": [
                "--trace",
                "--install"
            ],
```

**Comment by hellt** [comment]
The problem with a wrong `health` reporting seems to be stemming from the fact that telnet to vFPC doesn't work, hence vFPC status can't be checked

```
❯ telnet clab-vr02-vmx 5001
Trying 2001:172:20:20::3...
telnet: connect to address 2001:172:20:20::3: Connection refused
Trying 172.20.20.3...


^C
```

**Comment by monrad** [comment]
> I suppose this is due to the fact, that the image that is commited after the install phase has the `--trace --install` arguments in CMD
> 
> result of `image inspect`
> 
> ```
>      "Cmd": [
>                 "--trace",
>                 "--install"
>             ],
> ```
> 
> adding a [rewrite of the CMD](https://docs.docker.com/engine/reference/commandline/commit/#commit-a-container-with-new-cmd-and-expose-instructions) (in makefile-install) is a workaround

I think this might be a recent issue/change in docker.
I added a new vmx image and all my instances got rebuilt with cmd --trace and --install

**Comment by monrad** [comment]
I traced the CMDs not being removed to commit 5611463 I have created pull request #254 to reinstate the entrypoint.

**Comment by monrad** [comment]
The VFPC has changed its login prompt, I have created pull request #255 to fix this and the 4 versions I have tested show healthy now.

**Comment by mzagozen** [comment]
This was fixed with combination of #254 and #255. Thanks @monrad!
